### PR TITLE
fix: the errors of the lists

### PR DESCRIPTION
**Pull request by ardasnturk**
- added a control to `listHooks` to fix the problem of the first item's image not appearing in the `EventList` in text list type with large images
- fixed styles array format in `CardListItem`
- added `style` prop to `RNEImage` component to apply the style of the image more correctly
- updated the extra style of images in `TextListItem` created `withCard` prop and accepted as standard

SVA-1248

## Screenshots:

|before - list with large images|after - list with large images|before- list without large images|after - list without large images|
|--|--|--|--|
Simulator Screenshot - iPhone 14 Pro - 2023-12-15 at 13 46 40|Simulator Screenshot - iPhone 14 Pro - 2023-12-15 at 13 42 22|Simulator Screenshot - iPhone 14 Pro - 2023-12-15 at 13 48 10|Simulator Screenshot - iPhone 14 Pro - 2023-12-15 at 13 42 29

|before - event list|after - event list|before - text with small images|after - text with small images|
|--|--|--|--|
Simulator Screenshot - iPhone 14 Pro - 2023-12-15 at 14 12 44|Simulator Screenshot - iPhone 14 Pro - 2023-12-15 at 14 14 40|image|Simulator Screenshot - iPhone 14 Pro - 2023-12-15 at 14 13 42

### CODE REVIEW

File: src/components/CardListItem.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const renderCardContent = (item, horizontal) => {
         <Image
           source={{ uri: picture.url }}
           style={stylesWithProps({ horizontal }).image}
-          containerStyle={(styles.imageContainer, !!imageStyle && imageStyle)}
+          containerStyle={[styles.imageContainer, !!imageStyle && imageStyle]}
```
Comment (donni106) [inline src/components/CardListItem.js:26]: `containerStyle={[styles.imageContainer, imageStyle]}` you do not event have to check for the `imageStyle` in the array. if it is not defined, `containerStyle={[{...}, undefined]}` works totally fine

File: src/components/TextListItem.tsx [hunk 2]
```diff
@@ -107,15 +107,14 @@ export const TextListItem: NamedExoticComponent<Props> & {
...
-      {leftIcon ||
-        (leftImage && !!picture?.url ? (
-          <Image
-            source={{ uri: picture.url }}
-            style={[styles.smallImage, withCard && styles.withBigCardStyle]}
-            borderRadius={withCard ? normalize(8) : undefined}
-            containerStyle={styles.smallImageContainer}
-          />
-        ) : undefined)}
+      {(leftIcon || leftImage) && !!picture?.url ? (
```
Comment (donni106) [inline src/components/TextListItem.tsx:110]: the condition changed accidentally? if a `leftIcon` is give, it should be rendered. everything else was the "else" part, which is now changed?

File: src/components/TextListItem.tsx [hunk 3]
```diff
@@ -135,15 +134,14 @@ export const TextListItem: NamedExoticComponent<Props> & {
...
-      {rightIcon ||
-        (rightImage && !!picture?.url ? (
-          <Image
-            source={{ uri: picture.url }}
-            style={[styles.smallImage, withCard && styles.withBigCardStyle]}
-            borderRadius={withCard && normalize(8)}
-            containerStyle={styles.smallImageContainer}
-          />
-        ) : undefined)}
+      {(rightIcon || rightImage) && !!picture?.url ? (
```
Comment (donni106) [inline src/components/TextListItem.tsx:137]: the condition changed accidentally? if a rightIcon is given, it should be rendered. everything else was the "else" part, which is now changed?

File: src/hooks/listHooks.js [hunk 4]
```diff
@@ -102,7 +102,18 @@ export const useRenderItem = (query, navigation, options = {}) => {
           return <EventSectionHeader {...{ item, navigation, options, query }} />;
         }
 
-        if (index === 0) {
+        if (index === 0 || (query === QUERY_TYPES.EVENT_RECORDS && index === 1)) {
```
Comment (donni106) [inline src/hooks/listHooks.js:105]: i don't understand the case for the second entry in a event list having this special part? what about index 0 or 2 or 3?

File: src/components/CardListItem.js (hunk 1 above)
Comment (ardasnturk) [reply src/components/CardListItem.js:26]: Yes, you are right. I think the same applies to the other components. I'll fix them all.

`<RegularText
          small
          style={[!!generalStyle && generalStyle, !!overtitleStyle && overtitleStyle]}
        >` -> `<RegularText small style={[generalStyle, overtitleStyle]}>`

File: src/components/TextListItem.tsx (hunk 2 above)
Comment (ardasnturk) [reply src/components/TextListItem.tsx:110]: I'm sorry. I misunderstood the logic here. I'll undo the changes.

File: src/components/TextListItem.tsx (hunk 3 above)
Comment (ardasnturk) [reply src/components/TextListItem.tsx:137]: I'm sorry. I misunderstood the logic here. I'll undo the changes.

File: src/hooks/listHooks.js (hunk 4 above)
Comment (ardasnturk) [reply src/hooks/listHooks.js:105]: There is a special case in `EventList`. In EventList, there is a `SectionHeader` where we show the dates. `SectionHeader` is the 0th index of the list. If we want the first item to have a big picture, the first item shown for `EventList` is the item at index 1. That's why I added a logic in this way. I can explain this with the comment line.

Comment (donni106) [reply src/hooks/listHooks.js:105]: Thanks, now i got it! You explained it well, please add it as a comment in the code, yes.

File: src/components/TextListItem.tsx (hunk 2 above)
Comment (ardasnturk) [reply src/components/TextListItem.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/commit/4550329f16ccdef1de562c18775b5f594b9564b4

File: src/components/CardListItem.js (hunk 1 above)
Comment (ardasnturk) [reply src/components/CardListItem.js]: https://github.com/smart-village-solutions/smart-village-app-app/commit/4550329f16ccdef1de562c18775b5f594b9564b4

**Review by donni106 — APPROVED** [review]

File: src/hooks/listHooks.js (hunk 4 above)
Comment (ardasnturk) [reply src/hooks/listHooks.js:107]: https://github.com/smart-village-solutions/smart-village-app-app/pull/981/commits/3965aa1441ec9e3a14e581d5a80b6694310bc142

**Pull request merged**